packages/react-reconciler/src/ReactFiberNewContext.js

```diff
@@ -659,8 +697,20 @@ export function checkIfContextChanged(
       ? context._currentValue
       : context._currentValue2;
     const oldValue = dependency.memoizedValue;
-    if (!is(newValue, oldValue)) {
-      return true;
+    const compare = dependency.compare;
```

Comment:
should pull this inside the feature flag gate check and typecase to the variant that has the extra properties. This way builds that don't use this won't even have this property at all. This probably gets DCE'd so maybe it's fine outside but it'll remove cleaner if you just fork it